### ISSUE TITLE
Simulator: Integrate with new SimulatorBackend

### ISSUE THREAD

**Issue by MathewMacDougall** [UBC-Thunderbots/Software#1327]
### Description of the task

Once #1326 is complete we should be able to replace the old `SimulatorBackend` with the new Simulator system. The backend should only need to lightly wrap the available callback functions and mostly do data piping.

### Acceptance criteria

- [ ] Create a new SimulatorBackend

### Blocked By

#1326

**Comment by MathewMacDougall** [comment]
No longer relevant as changes are being made incrementally so the `SimulatorBackend` always works
